internal/core/command/device.go

```diff
@@ -104,14 +109,16 @@ func executeCommandByDevice(
 	body string,
 	queryParams string,
 	isPutCommand bool,
-	lc logger.LoggingClient) (string, error) {
+	lc logger.LoggingClient,
+	r *http.Request,
+	httpCaller internal.HttpCaller) (string, error) {
 
 	var ex Executor
 	var err error
 	if isPutCommand {
-		ex, err = NewPutCommand(device, command, body, ctx, &http.Client{}, lc)
+		ex, err = NewPutCommand(device, command, body, ctx, httpCaller, lc, r)
 	} else {
-		ex, err = NewGetCommand(device, command, queryParams, ctx, &http.Client{}, lc)
+		ex, err = NewGetCommand(device, command, queryParams, ctx, httpCaller, lc)
```

Comment:
I think we should add the same functionality for propagating headers for GET commands as well. This will make the service consistent. This will require adding tests for this as well.